src/stencils1d.jl

```diff
@@ -50,23 +50,30 @@ function get_stencil(
     xtarget;
     first = 1,
     extrapolate = Flat(),
+    reverse = false,
 )
     last = length(xsource)
-    if xtarget < xsource[1] # extrapolation
+    lowidx = reverse ? last : 1
+    highidx = reverse ? 1 : last
+
+    if xtarget < xsource[lowidx] # extrapolation
         # Linear extrapolation
-        extrapolate == LinearExtrapolation() && return (1, 2)
+        extrapolate == LinearExtrapolation() &&
+            (reverse ? (return (last - 1, last)) : (return (1, 2)))
         # Flat extrapolation by default
-        return (1, 1)
+        reverse ? (return (last, last)) : (return (1, 1))
```

Comment:
```suggestion
        return (lowidx, lowidx)
```

Reply:
It would be nice if we could do this for the `LinearExtrapolation` case too, to avoid checking `reverse` so many times. But I'm not sure how, since we need to subtract 1 in one case and add 1 in the other case, and the order is different

Reply:
I removed most of them!